components/org.wso2.carbon.identity.application.authenticator.fido2/src/test/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnServiceTest.java

```diff
@@ -0,0 +1,663 @@
+        when(mandatoryStages1.request(any(PublicKeyCredentialCreationOptions.class))).thenReturn(step2Finish);
+        when(step2Finish.response(any(PublicKeyCredential.class))).thenReturn(finishRegistrationOptionsBuilder);
+        when(finishRegistrationOptionsBuilder.build()).thenReturn(finishRegistrationOptions);
+
+        when(relyingParty.finishRegistration(any(FinishRegistrationOptions.class))).thenReturn(registrationResult);
+        PublicKeyCredentialDescriptor descriptor = mock(PublicKeyCredentialDescriptor.class);
+        ByteArray byteArray = new ByteArray(new byte[]{});
+        UserIdentity userIdentity = mock(UserIdentity.class);
+        when(registrationResult.getKeyId()).thenReturn(descriptor);
+        when(descriptor.getId()).thenReturn(byteArray);
+        when(registrationResult.getPublicKeyCose()).thenReturn(byteArray);
+        when(publicKeyCredentialCreationOptions.getUser()).thenReturn(userIdentity);
+        when(userIdentity.getId()).thenReturn(byteArray);
+        AuthenticatorSelectionCriteria authenticatorSelectionCriteria = mock(AuthenticatorSelectionCriteria.class);
+        when(publicKeyCredentialCreationOptions.getAuthenticatorSelection()).thenReturn(
+                Optional.ofNullable(authenticatorSelectionCriteria));
+        when(authenticatorSelectionCriteria.isRequireResidentKey()).thenReturn(requireResidentKey);
+
+        mockStatic(RegisteredCredential.class);
+        RegisteredCredential registeredCredential = mock(RegisteredCredential.class);
+        RegisteredCredential.RegisteredCredentialBuilder registeredCredentialBuilder =
+                mock(RegisteredCredential.RegisteredCredentialBuilder.class);
+        RegisteredCredential.RegisteredCredentialBuilder.MandatoryStages mandatoryStages2 =
+                mock(RegisteredCredential.RegisteredCredentialBuilder.MandatoryStages.class);
+        RegisteredCredential.RegisteredCredentialBuilder.MandatoryStages.Step2 step2RegisteredCredentials =
+                mock(RegisteredCredential.RegisteredCredentialBuilder.MandatoryStages.Step2.class);
+        RegisteredCredential.RegisteredCredentialBuilder.MandatoryStages.Step3 step3RegisteredCredentials =
+                mock(RegisteredCredential.RegisteredCredentialBuilder.MandatoryStages.Step3.class);
+
+        when(RegisteredCredential.builder()).thenReturn(mandatoryStages2);
+        when(mandatoryStages2.credentialId(any(ByteArray.class))).thenReturn(step2RegisteredCredentials);
+        when(step2RegisteredCredentials.userHandle(any(ByteArray.class))).thenReturn(step3RegisteredCredentials);
+        when(step3RegisteredCredentials.publicKeyCose(any(ByteArray.class))).thenReturn(registeredCredentialBuilder);
+        when(registeredCredentialBuilder.signatureCount(anyLong())).thenReturn(registeredCredentialBuilder);
+        when(registeredCredentialBuilder.build()).thenReturn(registeredCredential);
+
+        webAuthnService.finishFIDO2Registration(finishRegistrationResponseString);
```

Comment:
Don't we need an assertion for this as well?

Reply:
We cannot have a assertion here as there's no return value. Expectation is it should NOT throw any exception.